specification/imds/data-plane/Microsoft.InstanceMetadataService/stable/2018-10-01/imds.json

```diff
@@ -0,0 +1,731 @@
+        },
+        "ipv4Properties": {
+            "type": "object",
+            "properties": { 
+                "privateIpAddress": {
+                    "type":"string",
+                    "description":"This is the private IP address assigned to the interface."
+                 },
+                "publicIpAddress": { 
+                    "type":"string",
+                    "description":"This is the public IP address assigned to the interface."
+                }
+            }
+        },
+        "ipv6Properties": {
+            "type": "object",
+            "properties": { 
+                "privateIpAddress": {
+                    "type":"string",
+                    "description":"This is the private IPV6 address assigned to the interface."
+                }
+            }
+        },
+        "subnetProperties": {
+            "type": "object",
+            "properties": { 
+                "address": {
+                    "type":"string",
+                    "description":"This is the address range of the subnet."
+                },
+                "prefix": { 
+                    "type":"string",
+                    "description":"This is the prefix of the subnet."
+                }
+            }
+        },
+        "instance": {
+            "type": "object",
+            "properties": {
+                "compute": {
+                    "$ref": "#/definitions/compute",
+                    "description": "Compute Metadata"
+                },
+                "network": {
+                    "$ref": "#/definitions/network",
+                    "description": "Network Metadata" 
+                }
+            }
+        },
+        "attested": {
+            "type": "object",
+            "properties": {
+                "signature": {
+                    "type": "string",
+                    "description": "This is the encoded string containing the VM ID, plan information, and nonce value."
+                },
+                "encoding": {
+                    "type": "string",
+                    "description": "This is the encoding scheme of the signature."               
+                }
+            }
+        },
+        "ErrorResponse": {
+            "type": "object",
+            "properties": {
+                "error": {
+                    "type": "string",
+                    "description": "Error code",
+                    "enum": [
+                        "invalid_request",
+                        "unauthorize_client",
+                        "access_denied",
+                        "unsupported_response_type",
+                        "invalid_scope",
+                        "server_error",
+                        "service_unavailable",
+                        "bad_request",
+                        "forbidden",
+                        "not_found",
+                        "method_not_allowed",
+                        "too_many_requests"
+                    ]
```

Comment:
Please add the extension ["x-ms-enum"](https://github.com/Azure/autorest/tree/master/docs/extensions#x-ms-enum). I think the extension will have `modelAsString: true`. This will ensure that an expandable enum is created in strongly typed languages, thereby avoiding breaking changes when a new value is added/removed to/from the expandable enum.